### PR TITLE
Support setting `HostNetwork` and `DnsPolicy` using helm chart values

### PR DESCRIPTION
**Pull request by kahirokunn**
Issue #, if available:

Description of changes:
- Support setting `HostNetwork` and `DnsPolicy` using helm chart values

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by kahirokunn** [comment]
I just want to do this.
https://github.com/aws-controllers-k8s/acm-controller/pull/20/files

**Comment by kahirokunn** [comment]
@a-hilaly Hi!
I would like to contribute to the many repositories of aws-controllers-k8s, can you put me in the org?

File: templates/config/controller/deployment.yaml.tpl [hunk 1]
```diff
@@ -81,3 +81,12 @@ spec:
...
       hostPID: false
+      {{ if .Values.hostNetwork }}
+      hostNetwork: true
+      dnsPolicy: ClusterFirstWithHostNet
+      {{ else }}
+      hostNetwork: false
+      {{ with .Values.dnsPolicy }}
+      dnsPolicy: {{ . }}
+      {{ end }}
+      {{ end }}
```
Comment (a-hilaly) [inline templates/config/controller/deployment.yaml.tpl:92]: This is not a helm template, we can hard code the default values :)

File: templates/helm/templates/deployment.yaml [hunk 2]
```diff
@@ -143,6 +143,14 @@ spec:
       hostIPC: false
       hostNetwork: false
       hostPID: false
+      {{ if .Values.hostNetwork }}
```
Comment (a-hilaly) [inline templates/helm/templates/deployment.yaml:146]: This should be `.Values.deployment.hostNetwork`

File: templates/helm/templates/deployment.yaml [hunk 3]
```diff
@@ -143,6 +143,14 @@ spec:
...
+      hostNetwork: true
+      dnsPolicy: ClusterFirstWithHostNet
+      {{ else }}
+      hostNetwork: false
+      {{ with .Values.dnsPolicy }}
```
Comment (a-hilaly) [inline templates/helm/templates/deployment.yaml:151]: This should be `.Values.deployment.dnsPolicy`

**Review by a-hilaly — COMMENTED** [review]
Hi @kahirokunn - Thank you for your contribution!

 I left few comments in-line.

/ok-to-test

**Comment by a-hilaly** [comment]
>@a-hilaly Hi!
I would like to contribute to the many repositories of aws-controllers-k8s, can you put me in the org?

Unfortunately, the invitation process is automated and only @aws members can also join this org.

**Comment by kahirokunn** [comment]
@a-hilaly 
I have addressed the points you made.We have addressed the points you raised. Thx.

File: templates/helm/values.yaml.tpl [hunk 4]
```diff
@@ -28,6 +28,10 @@ deployment:
   # Which priorityClassName to set?
   # See: https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#pod-priority
   priorityClassName: ""
+  hostNetwork: false
+  # Which dnsPolicy to set?
+  # See: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy
+  dnsPolicy:
```
Comment (a-hilaly) [inline templates/helm/values.yaml.tpl:34]: @kahirokunn  Looks like tests are failing. I suspect that this fields needs a default policy - what's the default DNS policy for pods in kubernetes?

File: templates/helm/templates/deployment.yaml [hunk 5]
```diff
@@ -141,8 +141,15 @@ spec:
...
-      hostNetwork: false
       hostPID: false
+      {{ if .Values.deployment.hostNetwork }}
+      hostNetwork: true
+      dnsPolicy: ClusterFirstWithHostNet
+      {{ else }}
+      hostNetwork: false
+      {{ with .Values.deployment.dnsPolicy }}
+      dnsPolicy: {{ . }}
+      {{ end }}
```
Comment (a-hilaly) [inline templates/helm/templates/deployment.yaml:152]: Based on the error logs from the prow job, it appears that there may be some formatting errors. I recommend generating a controller and testing it to verify if it functions correctly.

Comment (a-hilaly) [reply templates/helm/templates/deployment.yaml:152]: (not on a laptop right now, i can try to generate and give more concise feedback tomorrow)

Comment (a-hilaly) [reply templates/helm/templates/deployment.yaml:152]: not familiar with the `with` key word here, let's keep it simple and do something like
```suggestion
      {{ if .Values.deployment.hostNetwork }}
      hostNetwork: true
      dnsPolicy: ClusterFirstWithHostNet
      {{ else }}
      hostNetwork: false
      dnsPolicy: {{ Values.deployment.dnsPolicy }}
      {{ end }}
```

Comment (a-hilaly) [reply templates/helm/templates/deployment.yaml:152]: or maybe something like:
```yaml
      hostNetwork: {{ .Values.deployment.hostNetwork }}
      dnsPolicy: {{ .Values.deployment.dnsPolicy }}
```

Comment (kahirokunn) [reply templates/helm/templates/deployment.yaml]: Yes, that is true. I set a default value and it works, so that's what I'll do.

Comment (kahirokunn) [reply templates/helm/templates/deployment.yaml]: Thx

File: templates/helm/values.yaml.tpl [hunk 6]
```diff
@@ -28,6 +28,10 @@ deployment:
   # Which priorityClassName to set?
   # See: https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#pod-priority
   priorityClassName: ""
+  hostNetwork: false
+  # Which dnsPolicy to set?
+  # See: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy
+  dnsPolicy: ClusterFirst
```
Comment (a-hilaly) [inline templates/helm/values.yaml.tpl:34]:
```suggestion
  # Set DNS policy for the pod.
  # Defaults to "ClusterFirst".
  # Valid values are 'ClusterFirstWithHostNet', 'ClusterFirst', 'Default' or 'None'.
  # To have DNS options set along with hostNetwork, you have to specify DNS policy
  # explicitly to 'ClusterFirstWithHostNet'.
  dnsPolicy: ClusterFirst
```

**Review by a-hilaly — COMMENTED** [review]
Almost there, i have few suggestions below

**Review by a-hilaly — APPROVED** [review]
Thank you @kahirokunn !

**Review by jljaco — APPROVED** [review]
/lgtm

**Pull request merged**